src/data/ha-state-history-data.html

```diff
@@ -68,6 +68,17 @@
               }
             });
             return result;
+          }).filter((element, index, arr) => {
+            // Remove data point if it is equal to previous point and next point.
```

Comment:
**Question:**
Should we do this on the server-side? We could potentially save a lot of bandwidth for entities that frequently update with the same state. Also OK flagging that as a TODO and merging this code as-is for the immediate benefits.

Reply:
We shouldn't do this unconditionally as some other client/api might want to pull all data points.
A bandwidth-vs-cpu improvement would be to compress the huge json responses.

Another improvement would be to drop all the extra useless attributes server-side, however this would add knowledge about how the history is implemented to the backend.

Reply:
So here we compare only the attributes we know the history cares about. If we do general filtering in the backend we would have to compare all attributes which might be less effective

Reply:
Definitely agree, but I think a new endpoint specifically for graphs instead of general history querying doesn't seem unreasonable as a long-term plan. Even though it would be somewhat catered to our frontend, it's a generally useful request for other clients that want to display graphs as well.

Reply:
In any case we should take that design decision to Paulus.
Submitting this one as-is meanwhile.